server/mssql/procedure.sql

```diff
@@ -1,6 +1,72 @@
 use major
 go
 
+CREATE PROCEDURE UpdateUserDetailsByUsername
+    @inputUsername varchar(25),
+    @newFullname nvarchar(50),
+    @newEmail varchar(100),
+    @newPhoneNumber varchar(15),
+    @newAddress nvarchar(100),
+    @newDateOfBirth date
+AS
+BEGIN
+    -- Update informations in userDetails table
+    UPDATE userDetails
+    SET fullname = @newFullname,
+        email = @newEmail,
+        phoneNumber = @newPhoneNumber,
+        address = @newAddress,
+        dateOfBirth = @newDateOfBirth
+    FROM users
+    WHERE users.username = @inputUsername AND users.id = userDetails.id;
+END;
```

Comment:
duplicated: [PROCEDURE updateUserInformation](https://github.com/iamSlightlyWind/majors_choice/blob/82ddcbb20d5466d703b13921866551b8d3985c8c/server/mssql/procedure.sql#L478-L536): update information, if any field is not present then it will not be updated

Please use that instead